strax/dtypes.py

```diff
@@ -201,3 +205,61 @@ def peak_dtype(n_channels=100, n_sum_wv_samples=200, n_widths=11):
         (('Maximum interior goodness of split',
           'max_goodness_of_split'), np.float32),
     ]
+
+
+def copy_to_buffer(source: np.ndarray,
+                   buffer: np.ndarray,
+                   func_name: str,
+                   field_names: ty.Tuple[str] = None):
+    """
+    Copy the data from the source to the destination e.g. raw_records to
+        records. To this end, we dynamically create the  njitted function
+        with the name 'func_name' (should start with "_").
+
+    :param source: array of input
+    :param destination: array of buffer to fill with values from input
+    :param func_name: how to store the dynamically created function.
+        Should start with an _underscore
+    :param field_names: dtype names to copy (if none, use all in the
+        source)
+    """
+    if np.shape(source) != np.shape(buffer):
+        raise ValueError('Source should be the same length as the buffer')
+
+    if field_names is None:
+        # To lazy to specify what to copy, just do all
+        field_names = tuple(n for n in source.dtype.names
+                            if n in buffer.dtype.names)
+    elif not any([n in buffer.dtype.names for n in field_names]):
+        raise ValueError('Trying to copy dtypes that are not in the '
+                         'destination')
+
+    if not func_name.startswith('_'):
+        raise ValueError('Start function with "_"')
+
+    if func_name not in globals():
+        # Create a numba function for us
+        _create_copy_function(buffer.dtype, field_names, func_name)
+
+    globals()[func_name](source, buffer)
+
+
+def _create_copy_function(res_dtype, field_names, func_name):
+    """Write out a numba-njitted function to copy data"""
+    # Cannot cache = True since we are creating the function dynamically
```

Comment:
I think the default of cache is True in numba is not it?

Reply:
Not sure but setting it to cache = True did raise problems. I think it's because there is no function for numba to inspect